### ISSUE TITLE
Why hasn't the fix or feature from release/1.5.x been synchronized to the master branch？

### ISSUE THREAD

**Issue by monkeyDluffy6017** [Kong/lua-resty-healthcheck#138]
Why hasn't these fix from release/1.5.x been synchronized to the master branch？
https://github.com/Kong/lua-resty-healthcheck/commit/c619d8c9cb9e6369204c4ed8ab26990198af9b3e
https://github.com/Kong/lua-resty-healthcheck/commit/19b23d8e47e9cdf92ec847387fc70cf2fa15e68a
https://github.com/Kong/lua-resty-healthcheck/commit/47f87093004aee15c02d38e42977dc7615cb140e
https://github.com/Kong/lua-resty-healthcheck/commit/e434007e8fe40e3894758907faa9d31db4227831

**Comment by nowNick** [comment]
Hi @monkeyDluffy6017 

Apologies for the confusion. The `master` branch is what's been extended since version `2.0.0`. Unfortunately because that version introduced breaking changes it was difficult for us to switch to it at kong. Therefore we kept using branch `1.y.z` which received fixes and updates. After some time kong changed the underyling library it uses from `lua-worker-events` to `lua-resty-events` which diverted even further from `2.0.0` version. 

Right now we are at the point where theoretically `master` is the head of repository with the latest version `2.0.0`, but that version has not been changed for 3 years. Whereas - as a matter of fact - our main branch is `1.6.x` and that's where those commits are. It produces a lot of confusion and it's time to clean this up.

To untangle all this mess we decided to release version `3.0.0` where we basically promote branch `1.6.x` with fixes that were introduced to master branch and we'll gradually port some of the fixes from `2.0.0` release. We'll also switch `master` branch from it's `2.0.0` version to the one that comes out of release of `3.0.0`. However version `2.0.0` will probably not be further developed. 

Hope that answers your question.
